File: packages/expo-drag-drop-content-view/ios/ExpoDragDropContentViewModule.swift
import ExpoModulesCore

let IOnDropEvent = "onDrop"
let IOnDragStartEvent = "onDragStart"
let IOnDragEndEvent = "onDragEnd"
let IOnExitEvent = "onExit"
let IOnEnterEvent = "onEnter"


public class ExpoDragDropContentViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoDragDropContentView")


    let fileSystem: EXFileSystemInterface? = self.appContext?.fileSystem

    View(ExpoDragDropContentView.self) {
        Events(IOnDropEvent, IOnDragStartEvent, IOnDragEndEvent, IOnEnterEvent, IOnExitEvent)

        Prop("includeBase64") { (view, includeBase64: Bool?) in
            let include = includeBase64 ?? false
            view.dragDropContentView.setIncludeBase64(include)
            view.dragDropContentView.fileSystem = fileSystem
        }
        Prop("draggableSources") { (view, draggableSources: [DraggableSource]?) in
            let sources = draggableSources ?? []
            view.dragDropContentView.setDraggableSources(sources)
        }
        Prop("mimeTypes") { (view, mimeTypes: [String: String]?) in
            let types = mimeTypes ?? [:]
            view.dragDropContentView.setMimeTypes(types)
        }
        Prop("allowedMimeTypes") { (view, allowedMimeTypes: [String]?) in
            view.dragDropContentView.setAllowedMimeTypes(allowedMimeTypes)
        }
    }
  }
}

File: packages/expo-drag-drop-content-view/android/src/main/java/expo/modules/dragdropcontentview/ExpoDragDropContentViewModule.kt
package expo.modules.dragdropcontentview

import expo.modules.dragdropcontentview.records.DraggableItem
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoDragDropContentViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoDragDropContentView")

    // Enables the module to be used as a native view. Definition components that are accepted as part of
    // the view definition: Prop, Events.
    View(ExpoDragDropContentView::class) {
      Prop("includeBase64") {view: ExpoDragDropContentView, value: Boolean? ->
        view.setIncludeBase64(value)
      }
      Prop("draggableSources") {view: ExpoDragDropContentView, value: List<DraggableItem>? ->
         view.setDraggableSources(value)
      }
      Prop("allowedMimeTypes") {view: ExpoDragDropContentView, value: List<String>? ->
         view.setAllowedMimeTypes(value)
      }
      Events("onDrop", "onDropListeningStart", "onDragStart", "onDragEnd", "onEnter", "onExit")
    }
  }
}
